Hotel 3.4

dudarenko-dmitry opened the pull request:
TASK_3.4: Removed TASK_3.4 to its own new branch Project_Hotel.
In Project_Hotel was created new branch Hotel_3.4.

dudarenko-dmitry committed:
add constructor and getter in Number.
create business-logic in Main

dudarenko-dmitry committed:
Task 3.2 is done

dudarenko-dmitry committed:
Merge pull request #2 from dudarenko-dmitry/TASK_3.1

Task_3.1 is done

dudarenko-dmitry committed:
Created abstract class Flower and add new classes which extend it.
Created interface Bouquet and BouquetImpl which implements it.
Rewrote class Main.

dudarenko-dmitry committed:
Merge pull request #3 from dudarenko-dmitry/TASK_3.2

Task 3.2 is done

dudarenko-dmitry committed:
Initial commit for branch TASK_3.4.
Added Hotel's Class diagram.

dudarenko-dmitry committed:
Created Entities' classes.

dudarenko-dmitry committed:
Created interfaces for Controller, Service, Repository layers.

dudarenko-dmitry committed:
TASK_3.4:
Created DataStorage package and Classes for all Entities.
Created Controller, Service and Repository layer for Room.
Added String constants for RoomService.
Added String constants for Room in Console.

dudarenko-dmitry committed:
TASK_3.4:
Created Controller, Service and Repository layer for Client.
Added String constants for ClientService.
Added String constants for Client in Console.

dudarenko-dmitry committed:
TASK_3.4:
Add attribute Client to HotelService.java;

dudarenko-dmitry committed:
TASK_3.4:
Change some attributes in HotelService, RoomReservation, Restorant, GuideTour.
Created Controller, Service, Repository layers for RoomReservation.

dudarenko-dmitry committed:
Created new branch Project_Hotel (as main branch for Task_Hotel) and Hotel_3.4 (for TASK_3.4).
Merged TASK_3.4 to Hotel_3.4.

mrseneca commented:
Why there some file for 3.1 and 3.2 subtasks? I think it appears here because of some mistake, please check it

dudarenko-dmitry committed:
Hotel_3.4:
Add to RoomReservation checking if using Client and Room are available.

dudarenko-dmitry commented:
> 
Mark wrote in Skype:
@все some words about 4 homework. It will be your long-term project which you will write/populate/rewrite during all following classes so please create a new clear branch for this and ONLY after approving task 3.4 move it to the new branch, commit it to there, and after that create a new branch right after committed changes. Here you will do your 4 homework and after that create a pull request into it and call it Task 4 or Homework 4, it's up to you

I did these operations:
1. Created in Master new branch Project_Hotel (as a Main/Develop branch for new project).
2. Created in Project_Hotel new branch Hotel_3.4 and merged into this new branch previous branch Lecture3/Task_3.4.
3. In previous branch Lecture3/Task_3.4 existed all other subtasks from Lecture3 (3.1 and 3.2).

What did I do wrong? 
Should I delete package Lecture3 from new Hotel_3.4 to make it more readable?

dudarenko-dmitry commented:
Should I also delete pull-request "TASK_3.4"?
New PR "Hotel_3.4" repeats all previous commits from "TASK_3.4" and includes brand new commits.

dudarenko-dmitry committed:
Hotel_3.4:
Add to RoomReservation's Update and Delete methods logic with changing Room Status and Cost.

dudarenko-dmitry committed:
Hotel_3.4:
Created Controller, Service and Repository layers for entity Order.
